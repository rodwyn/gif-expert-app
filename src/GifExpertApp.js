import React, { useState } from 'react';

const GiftExpertApp = () => {
  const [categories, setCategories] = useState(['Gorillaz', 'Deftones', 'The Chemical Brothers' ]);
  const handleAdd = () => {
    setCategories(['Radiohead', ...categories]);
  };

  return (
    <>
      <h2>GiftExpertApp</h2>
      <hr />

      <button onClick={handleAdd}>Agregar</button>

      <ol>
        {
          categories.map(category => {
            return <li key={category}>
              {category}
            </li>
          })
        }
      </ol>
    </>
  );
};

export default GiftExpertApp;