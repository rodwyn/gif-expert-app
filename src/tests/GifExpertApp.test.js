import React from 'react';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';


describe('Testing the <GifExpertApp /> component.', ()=>{
  test('Should render the component correctly.', () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Should render a category list.', () => {
    const categories = ['gorillaz', 'radiohead'];
    const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);

    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  })
  
});