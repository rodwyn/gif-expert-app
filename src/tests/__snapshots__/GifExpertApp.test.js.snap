// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the <GifExpertApp /> component. Should render the component correctly. 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
